content/applications/sales/sales/invoicing/milestone.rst

```diff
@@ -2,32 +2,241 @@
+
+When :guilabel:`Project & Task` or :guilabel:`Project` is selected, two new fields appear:
+:guilabel:`Project Template` and :guilabel:`Workspace Template`.
 
-You have to set the product type as *Service* under general
-information and select *Milestones* in the sales tab.
+.. image:: milestone/project-task-option-project-workspace-fields.png
+   :align: center
+   :alt: The Project template and workspace template fields that appear on milestone product.
 
-.. image:: milestone/milestone01.png
-    :align: center
+The :guilabel:`Project Template` field provides template options to use for the project that will be
+created when this specific product is ordered.
+
+The :guilabel:`Workspace Template` field provides template options to use for the workspace (for the
+*Documents* app, not the *Projects* app) that will be automatically generated for the project when
+this specific product is ordered.
+
+.. tip::
+   For organizational purposes, click the :guilabel:`Sales` tab on the product form, and enter a
+   custom 'Milestone' related descriptor in the :guilabel:`Sales Description` field. This
+   information appears in the :guilabel:`Description` column on the :guilabel:`Order Lines` tab of
+   the sales order.
+
+   Or, directly edit/modify the :guilabel:`Description` field on the :guilabel:`Order Lines` tab of
+   the sales order.
+
+   This is *not* a requirement.
 
 Invoice milestones
 ==================
 
-From the sales order, you can manually edit the quantity delivered as
-you complete a milestone.
+.. note::
+   The following flow features a trio of milestone products that have :guilabel:`Service` set as
+   their :guilabel:`Product Type`, and :guilabel:`Task` set on their :guilabel:`Create on Order`
+   field.
+
+    .. image:: milestone/settings-for-workflow.png
+       :align: center
+       :alt: Product with Service "Product Type" and "Task" in the Create on Order field on form.
+
+    Those tasks are then attached to a pre-existing :guilabel:`Project`, which, in this case, is
+    titled, :guilabel:`Rebranding Projects`.
+
+To invoice milestones, create a sales order with the milestone product(s). To do that, go to
+:menuselection:`Sales app --> New`. Doing so reveals a blank quotation form.
+
+From this quotation form, add a :guilabel:`Customer`. Then, click :guilabel:`Add a product` in the
+:guilabel:`Order Lines` tab. Next, add the milestone product(s) to the :guilabel:`Order Lines` tab.
+
+Once the corresponding milestone product(s) have been added, click :guilabel:`Confirm` to confirm
+the order, which turns the quotation into a sales order.
+
+When the order is confirmed, new smart buttons appear at the top of the sales order based on what
+was selected in the :guilabel:`Create on Order` field on the product form.
+
+.. image:: milestone/confirmed-sales-order-milestone.png
+   :align: center
+   :alt: Confirmed sales order with milestone products in the order lines tab.
```

Comment:
Ideally this image should be cropped, there is no need to show the whole UI here and the smartbuttons are illegible. Could even remove this image :)
```suggestion
```